### PR TITLE
Re-enabling counting API downloads/creating guestbookresponses

### PR DESCRIPTION
**Pull request by landreev**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #3331 : Guestbook - Downloads via API are not counted

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/4.6.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java [hunk 1]
```diff
@@ -183,12 +183,12 @@ public DownloadInstance datafile(@PathParam("fileId") Long fileId, @QueryParam("
         DataFile df = dataFileService.find(fileId);
         GuestbookResponse gbr = null;    
         
-        /*
+        
         if (gbrecs == null && df.isReleased()){
-            //commenting out for 4.6 SEK
-           // gbr = guestbookResponseService.initDefaultGuestbookResponse(df.getOwner(), df, session);
+            // commenting out for 4.6 SEK - ??
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Access.java]: I mentioned to @sekmiller this morning that if he's ok with it, we should simply remove this "commenting out" comment.

**Comment by sekmiller** [comment]
Looks good. I will pass to QA. One thing I noticed is that we seem to suppress the download URL on the file landing page if there is a Guestbook present. Probably should still do so in order to try to force the user to go through the interface for downloading. If download is done via the api while a GB is present on the dataset then any custom questions are ignored and required fields are not validated.

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java (hunk 1 above)
Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/api/Access.java]: I changed the comment to explain what we're doing here - that is writing a default gb response if the response hasn't been written via the interface

**Comment by pdurbin** [comment]
@sekmiller good point. #2911 (File API download bypasses terms of use) is highly related to all of this. I'd still like to have that comment removed, if you don't mind. This one: `// commenting out for 4.6 SEK - ??`

**Comment by pameyer** [comment]
As of 140c4ccbbc12a9c5adce4c47f1c6e94925ec4fa7 

Failed tests:   testCreateDataverseCreateDatasetUploadFileDownloadFileEditTitle(edu.harvard.iq.dataverse.api.SwordIT): expected:<404> but was:<500>

**Pull request merged**